### PR TITLE
[Android] MediaStore Support

### PR DESCRIPTION
**Pull request by ian-wd**
Added MediaStore support for android
- Only supports API lvl 29 and higher

Inspired by https://github.com/RonRadtke/react-native-blob-util/wiki/MediaStore/

### CODE REVIEW

File: example/App/example3.tsx [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        dummyImagePath,
+      );
+
+      setImageURI(contentURI);
+
+      const contentResult = await RNFS.readFile(contentURI, 'base64');
+
+      setImage(`data:image/png;base64,${contentResult}`);
+    } catch (err) {
+      console.log(err);
```
Comment (iBotPeaches) [inline example/App/example3.tsx:151]: Should we do `console.err`?

File: example/App/example3.tsx [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+    // get file stat to ensure file is not corrupted
+    // and to add another layer of check if RNFS.exists() fails to return the true value
+    try {
+      const fileStat = await RNFS.stat(destinationPath);
+      if (fileStat?.size > 0 && fileStat?.isFile()) {
+        return destinationPath;
+      }
+    } catch {
+      console.log('File does not exist');
```
Comment (iBotPeaches) [inline example/App/example3.tsx:96]: `err` as well

File: example/App/example3.tsx [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+          name: `${prefix}${imageFileName}`,
+          parentFolder: 'RNFSExample3Folder',
+          mimeType: 'image/png',
+        },
+        RNFS.MediaStore.MEDIA_IMAGE,
+        imagePath,
+      );
+
+      // TODO return content URI
+      return contentURI;
```
Comment (iBotPeaches) [inline example/App/example3.tsx:49]: This todo looks valid already?

**Review by iBotPeaches — COMMENTED** [review]
A lot of this code is shorthand. Can we swap to `{}` for all if/else/etc? Its tough to read for me with all this fancy single liners.

File: android/src/main/java/com/rnfs2/Utils/FileDescription.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package com.rnfs2.Utils;
+
+public class FileDescription {
+  public String name;
+  public String partentFolder;
```
Comment (iBotPeaches) [inline android/src/main/java/com/rnfs2/Utils/FileDescription.java:5]: Missed this first time. This is probably `parentFolder` no?

File: android/src/main/java/com/rnfs2/Utils/MimeType.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      String mt = MimeTypeMap.getSingleton().getMimeTypeFromExtension(fileExtension);
+      if (mt != null) return mt;
+      else return UNKNOWN;
+    }
+  }
+
+  /**
+   *
+   */
+  public static String stripEnd(String str, String stripChars) {
```
Comment (iBotPeaches) [inline android/src/main/java/com/rnfs2/Utils/MimeType.java:72]: Empty docblock

**Review by iBotPeaches — APPROVED** [review]

**Review by nemanjar7 — APPROVED** [review]

**Review by jyann — APPROVED** [review]

**Pull request merged**